lib/src/screens/support_chat/chatwoot_widget.dart

```diff
@@ -0,0 +1,85 @@
+import 'dart:convert';
+
+import 'package:flutter/material.dart';
+import 'package:flutter_inappwebview/flutter_inappwebview.dart';
+import 'package:flutter_secure_storage/flutter_secure_storage.dart';
+
+const COOKIE_KEY = 'chatwootCookie';
+
+class ChatwootWidget extends StatefulWidget {
+  ChatwootWidget(this.secureStorage,
+      {required this.baseUrl,
+      required this.websiteToken,
+      this.locale = "en",
+      this.authToken = ""});
+
+  final FlutterSecureStorage secureStorage;
+  final String baseUrl;
+  final String websiteToken;
+  final String locale;
+  final String authToken;
+
+  @override
+  ChatwootWidgetState createState() => ChatwootWidgetState();
+}
+
+class ChatwootWidgetState extends State<ChatwootWidget> {
+  ChatwootWidgetState() : _webViewkey = GlobalKey();
+
+  GlobalKey _webViewkey;
+
+  @override
+  void initState() {
+    super.initState();
+    _webViewkey = GlobalKey();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    var chatwootUrl =
+        "${widget.baseUrl}/widget?website_token=${widget.websiteToken}&locale=${widget.locale}";
+
+    if (widget.authToken.isNotEmpty)
+      chatwootUrl += "&cw_conversation=${widget.authToken}";
+
+    return InAppWebView(
```

Comment:
ok, so `InAppWebView` only supports Mobile platforms, so this would throw an exception on Desktop platforms.

I would suggest moving all the logic (creating url, fetching cookies, etc...) to a separate ViewModel and decide before going to `support_chat_page` whether to go to it (in case of Mobile) or launch the browser (in case of Desktop) in which cases you would have the URL ready from the ViewModel